core/src/main/java/org/apache/iceberg/BaseRowDelta.java

```diff
@@ -91,7 +91,8 @@ public RowDelta validateNoConflictingAppends(Expression newConflictDetectionFilt
   protected void validate(TableMetadata base) {
     if (base.currentSnapshot() != null) {
       if (!referencedDataFiles.isEmpty()) {
-        validateDataFilesExist(base, startingSnapshotId, referencedDataFiles, !validateDeletes);
+        validateDataFilesExist(
+            base, startingSnapshotId, referencedDataFiles, !validateDeletes, conflictDetectionFilter);
```

Comment:
This does make the assumption that the conflict detection filter and referenced data files are related.

Reply:
I think that is a correct assumption as the conflict detection filter is our scan condition and referenced data files are data files were read.

Reply:
I think that's a valid assumption.